src/api/app/lib/backend/proxy_request.rb

```diff
@@ -0,0 +1,136 @@
+module Backend
+  # The backend shares (a subset of) our API routes. This is a service to proxy requests to those routes.
+  # It caches incoming/outgoing files in our tmp directory or makes use of the various forward/rewrite/redirect
+  # plugins for apache, nginx or lighttpd if configured.
+  class ProxyRequest
+    attr_accessor :path, :request
+
+    def initialize(request, path_override = nil)
+      @request = request
+      @path = path_override || path_from_request
+    end
+
+    def call
+      case request.method_symbol
+      when :get, :head
+        proxy_from_backend unless use_web_server_forward
+      when :post, :put, :delete
+        proxy_to_backend
+      end
+    end
+
+    private
+
+    def path_from_request
```

Comment:
could we have it tested? There isn't already gems to do this? I think even the standard lib can do something like that, right? https://docs.ruby-lang.org/en/2.7.0/URI/Generic.html#method-c-build

Reply:
All of this needs to be tested. Just not sure how yet as it depends on `ActionDispatch::Request` but not really on a controller test...